examples/jsm/renderers/webgpu/WebGPUUniformsGroup.js

```diff
@@ -133,6 +133,20 @@ class WebGPUUniformsGroup extends WebGPUBinding {
 
 	}
 
+	updateByType( uniform, offset ) {
```

Comment:
This was also a good inspiration! 😉 

I've refactored the UBO updates based on your code here: d5c3089f86ffda1d4b26d02687a691f104d95368

The idea is to encapsulate each uniform type into a new class which encodes the byte length and item size. Besides, `setValue()` can be used to set the actual value of the uniform at runtime. 

This means its now nicer to compute the total byte size of the UBO and the `offset` value required for writing at the correct position in the buffer can be computed automatically